reference-architectures/cloud_deploy_flow/README.md

```diff
@@ -0,0 +1,164 @@
+automation and oversight. The process starts with release validation, ensuring
+that only compliant builds reach the release stage. Rollout approvals then offer
+flexibility, allowing teams to implement either manual checks or automated
+responses depending on specific requirements.
+
+This setup provides a blueprint for organizations to streamline deployment
+cycles while maintaining robust governance. By using this demo, you can see how
+these components work together, from container build through deployment, in a
+way that minimizes disruption to existing processes and aligns with typical
+organizational change management practices.
+
+This demo showcases a complete workflow that begins with the build of a
+container and progresses through various stages, ultimately resulting in the
+deployment of a new application.
+
+## Overview of the Demo
+
+This demo illustrates the end-to-end deployment process, starting from the
+container build phase. Here's a high-level overview of the workflow:
+
+1.  **Container Build Process**: The demo begins when a container is built in
+    Cloud Build. Upon completion, a notification is sent to a Pub/Sub message queue.
+
+2.  **Release Logic**: A Cloud Function subscribes to this message queue,
+    assessing whether a release should be created. If a release is warranted, a
+    message is sent to a "Command Queue" (another Pub/Sub topic).
+
+3.  **Creating a Release**: A dedicated function listens to the "Command Queue"
+    and communicates with Cloud Deploy to create a new release. Once the release
+    is created, a notification is dispatched to the Pub/Sub Operations topic.
+
+4.  **Rollout Process**: Another Cloud Function picks up this notification and
+    initiates the rollout process by sending a `createRolloutRequest` to the
+    "Command Queue."
+
+5.  **Approval Process**: Since rollouts typically require approval, a
+    notification is sent to the `cloud-deploy-approvals` Pub/Sub queue. An approval
+    function then picks up this message, allowing you to implement your custom logic
+    or utilize the provided Website Demo to return JSON, such as
+    `{ "manualApproval": "true" }`.
+
+6.  **Deployment**: Once approved, the rollout proceeds, and the new application
+    is deployed.
+
+![Workflow Diagram](architecture.svg)
+
+## Getting Started
+
+To run this demo, follow these steps:
+
+1.  **Fork and Clone the Repository**:
+    Start by forking this repository to your GitHub account (So you can connect
+    GCP to this repo), then clone it to your local environment. After cloning,
+    change your directory to the deployment demo:
+
+    ```bash
+    cd platform-engineering/reference-architectures/cloud-deploy-flow
+    ```
+
+2.  **Set Up Environment Variables**:
+   Define the environment variables of which details can be found in in
+   [variables.tf](variables.tf) to match your Google Cloud project and GitHub configuration:
+
+3.  **Initialize and Apply Terraform**:
```

Comment:
In order to execute your Terraform we needed the following APIs already enabled:

* compute.googleapis.com
* iam.googleapis.com
* cloudresourcemanager.googleapis.com

add to prereqs